qiskit/transpiler/passes/synthesis/unitary_synthesis.py

```diff
@@ -635,11 +636,17 @@ def _decomposer_2q_from_target(self, target, qubits, approximation_degree):
         decomposers = []
 
         def is_supercontrolled(gate):
-            kak = TwoQubitWeylDecomposition(Operator(gate).data)
+            try:
+                kak = TwoQubitWeylDecomposition(Operator(gate).data)
+            except QiskitError:
+                return False
```

Comment:
I think it's just the `Operator(gate)` call that should be permitted to fail here, right?  Can we just try/catch that instead, so we don't accidentally hide errors raised by `TwoQubitWeylDecomposition`?

Reply:
That's fair it is just the operator construction that we need to catch here. I'll update this accordingly.

Reply:
Updated in: https://github.com/Qiskit/qiskit-terra/pull/9677/commits/2a066bc12d33b9130bb1af381df78dc530508e37